libbeat/processors/actions/copy_fields.go

```diff
@@ -0,0 +1,106 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package actions
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+
+	"github.com/elastic/beats/libbeat/beat"
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/elastic/beats/libbeat/processors"
+)
+
+type copyFields struct {
+	config copyFieldsConfig
+}
+
+type copyFieldsConfig struct {
+	Fields        []fromTo `config:"fields"`
+	IgnoreMissing bool     `config:"ignore_missing"`
+	FailOnError   bool     `config:"fail_on_error"`
+}
+
+func init() {
+	processors.RegisterPlugin("copy_fields",
+		configChecked(newCopyFields,
+			requireFields("fields"),
+		),
+	)
+}
+
+func newCopyFields(c *common.Config) (processors.Processor, error) {
+	config := copyFieldsConfig{
+		IgnoreMissing: false,
+		FailOnError:   true,
+	}
+	err := c.Unpack(&config)
+	if err != nil {
+		return nil, fmt.Errorf("failed to unpack the configuration of copy processor: %s", err)
+	}
+
+	f := &copyFields{
+		config: config,
+	}
+	return f, nil
+}
+
+func (f *copyFields) Run(event *beat.Event) (*beat.Event, error) {
+	var backup common.MapStr
+	if f.config.FailOnError {
+		backup = event.Fields.Clone()
+	}
+
+	for _, field := range f.config.Fields {
+		err := f.copyField(field.From, field.To, event.Fields)
+		if err != nil && f.config.FailOnError {
+			logp.Debug("copy_fields", "Failed to copy fields: %s", err)
```

Comment:
I wonder if someone sets `FailOnError` if error should be an except and the user expects to see a log entry? At the same time I could see it overflowing the log with messages in case of bad events.

Reply:
One of the reasons I have chosen debug level is to stay consistent with `rename_fields` which has the same option and logs these errors on this level.

Reply:
I am fine with logging on a different level. However, if it's changed in this processor, I would do a follow-up in `rename_fields` to not to surprise users.

Reply:
Though question, @ruflin @kvch 

Maybe we should ask ourself, as a User how would I discover that we have something wrong with our processor configuration?

1. I don't see anything in the log?
2. I don't have any special tag on the event so I cannot filter on error in kibana.

So I think the only what to the user to know that something is wrong is they really know their data and the processor operating on it. :(

Should we go the route of having errors tag for that too?

Reply:
@kvch Agree that we should fix it everywhere. I'm good to sticking what we have in rename for now and tackle all the processors together.

@ph Should we have something like `ReportOnError` and if set to true, it sets `error.message`? I'm sure @webmat from ECS perspective will also have an opinion here :-)

Reply:
@ruflin I like the idea of adding `error.message` to the event if the processor encountered an error. If we all agree on it, I am adding it to both this processor and `rename_fields` processor.

Reply:
@ruflin @kvch sound good ++

Reply:
Yes, `error.message` for now is being used for pipeline errors, rather than userland errors. So it's alright if it goes there.